HakawaiTests/HKWMentionsPluginTests.m

```diff
@@ -993,6 +996,73 @@ - (CGFloat)heightForCellForMentionsEntity:(id<HKWMentionsEntityProtocol> _Null_u
+    });
+
+    it(@"prepend @", ^{
+        mentionsPlugin.controlCharactersToPrepend = [NSCharacterSet characterSetWithCharactersInString:@"@"];
+        textView.text = @"@First";
+        HKWMentionsAttribute *attribute = [HKWMentionsAttribute mentionWithText:@"FirstName LastName" identifier:@"1"];
+        [mentionsPlugin createMention:attribute cursorLocation:textView.text.length - 1];
+
+        expect(textView.text).to.equal(@"@FirstName LastName");
+        expect(attribute.mentionText).to.equal(@"@FirstName LastName");
+    });
+    
+    it(@"prepend Japanese ＠", ^{
+        mentionsPlugin.controlCharactersToPrepend = [NSCharacterSet characterSetWithCharactersInString:@"@＠"];
+        textView.text = @"＠リンクトイン";
+        HKWMentionsAttribute *attribute = [HKWMentionsAttribute mentionWithText:@"リンクトイン アメリカ" identifier:@"2"];
+        [mentionsPlugin createMention:attribute cursorLocation:textView.text.length - 1];
+
+        expect(textView.text).to.equal(@"＠リンクトイン アメリカ");
+        expect(attribute.mentionText).to.equal(@"＠リンクトイン アメリカ");
+    });
+    
+    it(@"prepend @ with multiple controlCharactersToPrepend", ^{
+        mentionsPlugin.controlCharactersToPrepend = [NSCharacterSet characterSetWithCharactersInString:@"@#"];
+        textView.text = @"@First";
+        HKWMentionsAttribute *attribute = [HKWMentionsAttribute mentionWithText:@"FirstName LastName" identifier:@"3"];
+        [mentionsPlugin createMention:attribute cursorLocation:textView.text.length - 1];
+
+        expect(textView.text).to.equal(@"@FirstName LastName");
+        expect(attribute.mentionText).to.equal(@"@FirstName LastName");
+    });
+
+    it(@"not prepand character when controlCharactersToPrepend is empty", ^{
+        mentionsPlugin.controlCharactersToPrepend = nil;
+        textView.text = @"@First";
+        HKWMentionsAttribute *attribute = [HKWMentionsAttribute mentionWithText:@"FirstName LastName" identifier:@"4"];
+        [mentionsPlugin createMention:attribute cursorLocation:textView.text.length - 1];
+
+        expect(textView.text).to.equal(@"FirstName LastName");
+        expect(attribute.mentionText).to.equal(@"FirstName LastName");
+    });
+    
+    it(@"ignore control character with attribute", ^{
+        mentionsPlugin.controlCharactersToPrepend = [NSCharacterSet characterSetWithCharactersInString:@"@#"];
+        textView.text = @"@First";
+        HKWMentionsAttribute *attribute = [HKWMentionsAttribute mentionWithText:@"FirstName LastName" identifier:@"5"];
+        [mentionsPlugin createMention:attribute cursorLocation:textView.text.length - 1];
+        
+        attribute.range = NSMakeRange(0, textView.text.length);
+        textView.text = [textView.text stringByAppendingString:@" test"];
+        [mentionsPlugin addMention:attribute];
+        
+        expect([mentionsPlugin mostRecentControlCharacterLocationInText:@"@FirstName LastName test"]).to.equal(NSNotFound);
+    });
+});
```

Comment:
please add a test for "selectedRange", where u place the cursor in one of these mentions entities with a control char prepended, and make sure the selected range is correct